### PR TITLE
show your love for nuka cola

### PR DESCRIPTION
**Pull request by cornercube**
## About The Pull Request
adds a trait that allows you to sense and derive pleasure from the act of drinking cola and hate healthy foods
adds a new flavor, NUKA

## Why It's Good For The Game
because nuka cola is so good, you can get addicted to it in FO1 and 2
adds more utility to nuka cola products as well

feel free to add sunset somewhere down the line which makes you hate NUKA

## Pre-Merge Checklist

- [ ] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

## Changelog

:cl:
add: new flavor NUKA
add: NUKA flavor to nuka-cola and variants
trait: nuka fiend, which causes you to gain mood from drinking nuka cola and lose mood from heating fruits 

/:cl:

### CODE REVIEW

File: code/datums/traits/good.dm [hunk 1]
```diff
@@ -259,6 +259,27 @@
 	var/mob/living/carbon/human/H = quirk_holder
 	H.update_sight()
 
+/datum/quirk/nukalover
+	name = "Nuka Fiend"
```
Comment (Looreydev) [inline code/datums/traits/good.dm:263]:
```suggestion
	name = "Nuka Friend"
```

**Review by Looreydev — CHANGES_REQUESTED** [review]

File: code/datums/traits/good.dm (hunk 1 above)
Comment (cornercube) [reply code/datums/traits/good.dm:263]: Fiend is intentional, as per CRANKs suggestion

File: code/modules/fallout/reagents/drinks.dm [hunk 2]
```diff
@@ -11,8 +11,12 @@
 	M.drowsyness = 0
 	M.AdjustSleeping(-40, FALSE)
 	M.adjust_bodytemperature(-5 * TEMPERATURE_DAMAGE_COEFFICIENT, BODYTEMP_NORMAL)
+	if(HAS_TRAIT(M, TRAIT_NUKA_LOVER))
+		M.adjustBruteLoss(-0.01)
+		M.adjustFireLoss(-0.01)
+	else
```
Comment (lolman360) [inline code/modules/fallout/reagents/drinks.dm:17]: you can remove this entire else it doesnt do anything
```suggestion
```

File: code/datums/traits/good.dm [hunk 3]
```diff
@@ -259,6 +259,28 @@
...
+	mob_trait = TRAIT_NUKA_LOVER
+	gain_text = "<span class='notice'>You want to buy the whole world a nuka-cola!</span>"
+	lose_text = "<span class='danger'>What's the big deal about nuka-cola?</span>"
+	medical_record_text = "Patient has an addiction to the soft drink Nuka-Cola. Somehow, their metabolism has adapted to the sugars and artifical flavorings."
+	
+/datum/quirk/nukalover/add()
+	var/mob/living/carbon/human/H = quirk_holder
+	var/datum/species/species = H.dna.species
+	species.liked_food |= NUKA
+	species.disliked_food |= FRUIT
```
Comment (lolman360) [inline code/datums/traits/good.dm:275]: make them dislike vegetables instead  it'd be funnier
```suggestion
	species.disliked_food |= VEGETABLES
```

**Pull request merged**